src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -605,13 +605,32 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                     _includeCteIds.Add(TableExpressionName(_tableExpressionCounter));
                     break;
                 case TableExpressionKind.IncludeUnionAll:
-                    StringBuilder.AppendLine("SELECT Sid1, IsMatch, IsPartial ");
+                    StringBuilder.Append("SELECT Sid1, IsMatch, IsPartial ");
+                    bool supportedSortParamExists = context.SupportedSortParamExists();
+                    if (supportedSortParamExists)
+                    {
+                        StringBuilder.AppendLine(", SortValue");
+                    }
+                    else
+                    {
+                        StringBuilder.AppendLine();
+                    }
+
                     StringBuilder.Append("FROM ").AppendLine(_cteMainSelect);
 
                     foreach (var includeCte in _includeCteIds)
                     {
                         StringBuilder.AppendLine("UNION ALL");
-                        StringBuilder.AppendLine("SELECT Sid1, IsMatch, IsPartial ");
+                        StringBuilder.Append("SELECT Sid1, IsMatch, IsPartial");
+                        if (supportedSortParamExists)
+                        {
+                            StringBuilder.AppendLine(", NULL as SortValue ");
```

Comment:
why 'NULL'?

Reply:
Because you inner join with a table which don't contain SortValue column